### PR TITLE
[Delivers #104865396] Ensure adding observer with reason creates an update comment

### PR DESCRIPTION
**Pull request by yozlet**
Also, adding an observer _without_ specifying a reason should not create a comment.

### CODE REVIEW

File: app/models/proposal.rb [hunk 1]
```diff
@@ -268,7 +268,7 @@ def create_new_observation(user, adder, reason)
     self.observers(true)
     # when explicitly adding an observer using the form in the Proposal page...
     if adder
-      if reason
+      if reason and !reason.blank?
```
Comment (jessieay) [inline app/models/proposal.rb]: about about `if reason && reason.present?`

**Comment by jessieay** [comment]
> adding an observer without specifying a reason should not create a comment.

do we have a test for this? If not, can we add one?

**Comment by yozlet** [comment]
@jessieay There was a test for a `nil` reason but not a blank one, so I've added that. And I didn't know about `present?`, thank you!

File: spec/models/proposal_spec.rb [hunk 2]
```diff
@@ -314,13 +314,24 @@
         end
       end
 
+      context "with a blank reason" do
+        let(:reason) { ' ' }
```
Comment (jessieay) [inline spec/models/proposal_spec.rb]: what about just assigning `reason` within the `it` block!? (I see that would be inconsistent with the rest of this file...so not necessary, but certainly an organizational pattern I think we should be moving toward!!

File: app/models/proposal.rb [hunk 3]
```diff
@@ -268,7 +268,7 @@ def create_new_observation(user, adder, reason)
     self.observers(true)
     # when explicitly adding an observer using the form in the Proposal page...
     if adder
-      if reason
+      if reason and reason.present?
```
Comment (jessieay) [inline app/models/proposal.rb]: I think we're mostly using `&&` rather than `and` in this project!

File: spec/models/proposal_spec.rb [hunk 4]
```diff
@@ -314,13 +314,24 @@
...
+
+        it 'does not add a comment' do
+          expect(proposal.comments).to be_empty
+          proposal.add_observer(observer_email, user, reason)
+          expect(proposal.comments).to be_empty
```
Comment (jessieay) [inline spec/models/proposal_spec.rb]: niiiice

File: spec/models/proposal_spec.rb [hunk 5]
```diff
@@ -314,13 +314,24 @@
         end
       end
 
+      context "with a blank reason" do
+        let(:reason) { ' ' }
+
+        it 'does not add a comment' do
```
Comment (jessieay) [inline spec/models/proposal_spec.rb]: double quotes! We are enforcing this starting today :)

**Comment by yozlet** [comment]
@jessieay Fixed!

**Comment by jessieay** [comment]
I think you will fail code climate because we are now enforcing double quotes...but otherwise code looks :ship:  !

**Comment by pkarman** [comment]
@yozlet I think codeclimate ignores spec files and will never finish on this PR.
Do you have some advice for how to QA this?

**Comment by yozlet** [comment]
@pkarman Ah, that explains those non-finishing CC runs, thanks.

To QA, check that
* when adding an observer to a proposal with a reason
   * A formatted comment is added, with most of the comment italicised except the reason, which is in `monospace`
* when adding an observer to a proposal with a blank reason
   * no comment is added

**Comment by pkarman** [comment]
Looks great.

**Pull request merged**